src/Featurizers/LagLeadOperatorFeaturizer.h

```diff
@@ -0,0 +1,246 @@
+                    for (std::uint32_t col = _horizon - 1; col < _horizon; --col) {
+                        if (
+                            (_index - _numPending - col + _offsets[row] < 0) ||
+                            (_numPending <= _offsets[row] - col)) {
+                            ret(static_cast<std::int32_t>(row), static_cast<std::int32_t>(_horizon - 1 - col)) = std::nan("");
+                        }
+                        else {
+                            ret(static_cast<std::int32_t>(row), static_cast<std::int32_t>(_horizon - 1 - col)) = *std::get<0>(_buffer.range(
+                                                                                                1, 
+                                                                                                static_cast<size_t>(static_cast<std::int64_t>(_buffer.size() - _numPending - col) + _offsets[row])
+                                                                                              ));
+                        }
+                    }
+                }
+                callback(ret);
+                _numPending--;
+            }
+        }
+    }
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \typedef       LagLeadOperatorEstimator
+///  \brief         Estimator that creates `LagLeadOperatorTransformer`.
+///
+template <typename InputT>
+using LagLeadOperatorEstimator           = Components::InferenceOnlyEstimatorImpl<LagLeadOperatorTransformer<InputT>>;
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// |
+// |  Implementation
+// |
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+template <typename T>
+LagLeadOperatorTransformer<T>::LagLeadOperatorTransformer(std::uint32_t horizon, std::vector<std::int64_t> offsets) :
+    _horizon(
+        std::move(
+            [&horizon](void) -> std::uint32_t & {
+                if (horizon == 0) {
+                    throw std::invalid_argument("Horizon cannot be 0!");
+                }
+                return horizon;
+            }()
+    )),
+    _offsets(
+        std::move(
+            [&offsets](void) -> std::vector<std::int64_t> & {
+                if (offsets.size() == 0) {
+                    throw std::invalid_argument("Offsets is empty!");
+                }
+                return offsets;
+            }()
+        )
+    ),
+    // if there are more than one offset provided, we need find the difference between max and min value to determine buffer size
+    // and if there's only one offset, we use the absolute value of it as the size
+    _buffer(Components::CircularBuffer<TargetType>(
```

Comment:
>Components::CircularBuffer<TargetType> [](start = 12, length = 38)

As written, this code creates a temporary `CircularBuffer` type and then calls `_buffer`'s move constructor. Remove the class name to call the standard constructor.